server/util-server.js

```diff
@@ -89,6 +90,52 @@ exports.pingAsync = function (hostname, ipv6 = false) {
     });
 };
 
+exports.mqttAsync = function (hostname, topic, okMessage, options = {}) {
+    return new Promise((resolve, reject) => {
+        const { port, username, password, interval = 20 } = options;
+
+        // Adds MQTT protocol to the hostname if not already present
+        if (!/^(?:http|mqtt)s?:\/\//.test(hostname)) {
+            hostname = "mqtt://" + hostname;
+        }
+
+        debug("MQTT connecting");
+
+        let client = mqtt.connect(hostname, {
+            port,
+            username,
+            password
+        });
+
+        client.on("connect", () => {
+            debug("MQTT subscribe topic");
+            client.subscribe(topic);
+        });
+
+        client.on("error", (error) => {
+            client.end();
+            reject(error);
+        });
+
+        client.on("message", (messageTopic, message) => {
+            if (messageTopic == topic) {
+                if (message.toString() === okMessage) {
+                    client.end();
+                    resolve(`Topic: ${messageTopic}; Message: ${message.toString()}`);
+                } else {
+                    client.end();
+                    reject(new Error(`Error; Topic: ${messageTopic}; Message: ${message.toString()}`));
+                }
+            }
+        });
+
+        setTimeout(() => {
```

Comment:
Potential memory leak. It is created per check and never stop.

Reply:
I've added clear timeout in case client is already ended. Can you please check to see if this will fix the memory leak issue.